index.html

```diff
@@ -3697,149 +3786,181 @@ <h3>The <dfn>ThingFilter</dfn> dictionary</h3>
       </p>
       <pre class="idl">
         dictionary ThingFilter {
-          DiscoveryMethod method = "directory";
-          USVString? url;
           object? fragment;
           <!-- DOMString? query; -->
         };
       </pre>
-      <p class = "ednote">
-        The `DOMString query` property was temporarily removed from
-        <a>ThingFilter</a>, until it is standardized in the WoT Discovery
-        task force.
-      </p>
       <p>
-        The <dfn>method</dfn> property represents the discovery type that should be used in the discovery process. The possible values are defined by the <a href="#the-discoverymethod-enumeration">DiscoveryMethod</a> enumeration.
+        The <dfn>fragment</dfn> property represents a template object used for matching property by property against discovered <a>Thing</a>s.
       </p>
-      <p>
-        The <dfn>url</dfn> property represents the URL of the target entity serving the discovery request, for instance the URL of a <a>Thing Directory</a> (if <a href="#dom-thingfilter-method">method</a> is {{DiscoveryMethod/[["directory"]]}}),
-        or the URL of a directly targeted <a>Thing</a> (if
-        <a href="#dom-thingfilter-method">method</a> is {{DiscoveryMethod/[["direct"]]}})
+      <p class = "ednote">
+        The |query:DOMString| property was temporarily removed from
+        <a>ThingFilter</a>, until it is standardized in the WoT Discovery
+        task force. It represented a query string accepted by the implementation,
+        for instance a SPARQL or JSON query. Support was to be implemented locally
+        in the <a>WoT Runtime</a> or remotely as a service in a <a>Thing Directory</a>.
       </p>
-      <p>
-        The <dfn>fragment</dfn> property represents a template object used for matching property by property against discovered <a>Thing</a>s.
+      <p class="ednote">
+        The |url:USVString| property was removed. It used to represent the target
+        entity serving the discovery request, for instance the URL of a
+        <a>Thing Directory</a>, or the URL of a directly targeted <a>Thing</a>,
+        but these are implemented by dedicated methods now.
       </p>
-      <!--p>
-        The <dfn>query</dfn> property represents a query string accepted by the implementation, for instance a SPARQL or JSON query. Support may be implemented locally in the <a>WoT Runtime</a> or remotely as a service in a <a>Thing Directory</a>.
-      </p-->
     </section>
 
     <section>
-      <h3>The <dfn>start()</dfn> method</h3>
+      <h3>The <dfn data-lt="generic-discovery">generic discovery</dfn> algorithm</h3>
```

Comment:
A question partly related to https://github.com/w3c/wot-scripting-api/issues/390: Can/should we specify the runtime's behavior if it encounters an "Unauthorized" response? This relates to the [Security Bootstrapping](https://w3c.github.io/wot-discovery/#exploration-secboot) feature in the Discovery specification.

Reply:
First, if such a step is explicit, should be in the algorithms of the methods, not in the meta-algorithms.
But this is still a variant of " if not allowed".
Dialogs spawned by the runtime (or other local decisions/mechanisms/policies) are not in the scope of the API spec. That is something that happens in the underlying platform, but might have an effect here under the common clause of "not allowed". For privacy reasons, specs usually don't detail how exactly this failed.

Reply:
I see, then we can consider this question resolved :)